File: Lancement d'un programme exec/exo5.py
import os
import sys

def main():
    while True:
        f = input("Fichier à ouvrir (ou 'quitter') : ")

        if f == "quitter":
            break

        pid = os.fork()

        if pid == 0:
            # Processus enfant
            os.execl("/usr/bin/emacs", "emacs", f)
            os._exit(0)
        
       

if __name__ == "__main__":
    main()

File: Lancement d'un programme exec/exo5.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

int main() 
{
    char f[100];

    while (1) 
    {
        printf("Fichier à ouvrir (ou 'quitter') : ");
        scanf("%s", f);

        if (strcmp(f, "quitter") == 0) 
        {
            break;
        }

        if (fork() == 0) 
        {
            
            execl("/usr/bin/emacs", "emacs", f, NULL);
            exit(0); 
        }
        
    }
    return 0;
}